src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -206,11 +206,22 @@ public boolean handleInput(@Nullable Screen screen, String input, boolean needsF
                 persistentMenuId = guid;
             }
 
-            if (screen instanceof EntityScreen) {
-                String breadcrumb = screen.getBreadcrumb(input, sandbox, getSessionWrapper());
-                persistentMenuHelper.addEntitySelection(persistentMenuId, breadcrumb);
+            /**
+             *  we don't want to show any hidden menus on the persistent menu and it's impossible
+             *  for us to tell based on selection index whether a menu is visible or not. Therefore
+             *  we are restricting to not showing anything on persistent menu except visible root menu options
+             *  if we are not respecing relevancy here.
```

Comment:
nit: typo "respecting"

Reply:
corrected in https://github.com/dimagi/formplayer/pull/1594/commits/cee21e73ca54d536d6011f0273174f75383c3425